src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs

```diff
@@ -94,7 +95,8 @@ protected virtual void SerializeToFile(OmniSharpProjectSnapshot projectSnapshot,
         // by the time we move the tempfile into its place
         using (var writer = tempFileInfo.CreateText())
         {
-            _serializer.Serialize(writer, projectSnapshot);
+            var projectRazorJson = new ProjectRazorJson(publishFilePath, projectSnapshot.InternalProjectSnapshot);
```

Comment:
❓is there ever a time when serializing the snapshot is correct?

Reply:
It results in a circular dependency, so I don't think so (or at least not any scenario I can think of)